runner/run_cloud_test.go

```diff
@@ -0,0 +1,115 @@
+package runner
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strconv"
+	"testing"
+	"text/template"
+
+	"github.com/coreruleset/go-ftw/config"
+	"github.com/coreruleset/go-ftw/ftwhttp"
+	"github.com/coreruleset/go-ftw/output"
+	"github.com/coreruleset/go-ftw/test"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/suite"
+)
+
+type runCloudTestSuite struct {
+	suite.Suite
+	cfg          *config.FTWConfiguration
+	ftwTests     []test.FTWTest
+	out          *output.Output
+	ts           *httptest.Server
+	dest         *ftwhttp.Destination
+	tempFileName string
+}
+
+func TestRunCloudTestSuite(t *testing.T) {
+	suite.Run(t, new(runCloudTestSuite))
+}
+
+func (s *runCloudTestSuite) SetupTest() {
+	s.newTestCloudServer()
+	s.out = output.NewOutput("normal", os.Stdout)
+}
+
+func (s *runCloudTestSuite) TearDownTest() {
+	s.ts.Close()
+}
+
+func (s *runCloudTestSuite) BeforeTest(_ string, name string) {
+	var err error
+
+	// if we have a destination for this test, use it
+	// else use the default destination
+	if s.dest == nil {
+		s.dest, err = ftwhttp.DestinationFromString(destinationMap[name])
+		s.NoError(err)
+	}
+
+	log.Info().Msgf("Using port %d and addr '%s'", s.dest.Port, s.dest.DestAddr)
+
+	// set up variables for template
+	vars := map[string]interface{}{
+		"TestPort": s.dest.Port,
+		"TestAddr": s.dest.DestAddr,
+	}
+
+	s.cfg = config.NewCloudConfig()
+	// get tests template from file
+	tmpl, err := template.ParseFiles(fmt.Sprintf("testdata/%s.yaml", name))
+	s.NoError(err)
+	// create a temporary file to hold the test
+	testFileContents, err := os.CreateTemp("testdata", "mock-test-*.yaml")
+	s.NoError(err, "cannot create temporary file")
+	err = tmpl.Execute(testFileContents, vars)
+	s.NoError(err, "cannot execute template")
+	// get tests from file
+	s.ftwTests, err = test.GetTestsFromFiles(testFileContents.Name())
+	s.NoError(err, "cannot get tests from file")
+	// save the name of the temporary file so we can delete it later
+	s.tempFileName = testFileContents.Name()
+}
+
+func (s *runCloudTestSuite) AfterTest(_ string, _ string) {
```

Comment:
AFAICT, AfterTest and TearDownTest do the same thing, except that AfterTest has arguments. I suggest to use TearDownTest because we already use that and don't use the arguments.

Reply:
You are right, removing func and moving cleanup to TearDown.